prometheus/go_collector.go

```diff
@@ -279,3 +279,8 @@ type memStatsMetrics []struct {
 	eval    func(*runtime.MemStats) float64
 	valType ValueType
 }
+
+type runtimeEnvVarsMetrics []struct { // I couldn't come up with a better name. Any suggestions?
```

Comment:
Nice! I see where you go with this. This way would totally work, but it's extra structures in many places. There is a way to simplify this by reusing generic flow, so by default adding appropriate regexes to `RuntimeMetricRules` in defaultGoCollectorOptions (also you can remove the weird dead code (comment)) - https://github.com/prometheus/client_golang/blob/6478f56935c73f23b38144dc33cae75ed10554ac/prometheus/go_collector_latest.go#L162

I think we can add that by default, then fix tests. We could even test if `WithoutGoCollectorRuntimeMetrics` can turn those off 🤗 WDYT?